core/controlplane/config/templates/cloud-config-etcd

```diff
@@ -355,6 +377,43 @@ write_files:
     content: {{$w.GzippedBase64Content}}
   {{- end }}
 {{- end }}
+{{if and (.AmazonSsmAgent.Enabled) (ne .AmazonSsmAgent.DownloadUrl "")}}
+  - path: "/opt/bin/ssm/install-ssm-agent.sh"
+    permissions: 0700
+    content: |
+      #!/bin/bash
+      set -eo pipefail
+
+      function finish {
+        if [ "$(pwd)" == "/opt/bin/ssm" ]; then
+          popd > /dev/null 2>&1
+        fi
+      }
+
+      trap finish EXIT
+
+      pushd /opt/bin/ssm > /dev/null 2>&1
+
+      TAR_FILE=ssm.linux-amd64.tar.gz
+      CHECKSUM_FILE="${TAR_FILE}.sha1"
+
+      curl --silent -L -o ${TAR_FILE} "{{ .AmazonSsmAgent.DownloadUrl }}"
+      if [[ -f "${CHECKSUM_FILE}" ]]; then
+        ( sha1sum --status -c "${CHECKSUM_FILE}" && exit 0 ) || true
```

Comment:
whats the role of `CHECKSUM_FILE`? 

1. it is generated from TAR_FILE itself at the end of the script. Usually signed checksums are downloaded from same location where tarball is, to verify integrity of a tarball
2. it is never used on first download
3. checksum check results are ignored